src/adapters/DebugProtocolAdapter.ts

```diff
@@ -44,6 +44,12 @@ export class DebugProtocolAdapter {
         });
     }
 
+    private connectionDeferred = defer<void>();
+
+    public deferredConnectionPromise(): Promise<void> {
```

Comment:
Maybe call it something more like this? The word 'deferred' isn't necessary here, that's an internal implementation detail
```suggestion
    public isConnected(): Promise<void> {
```